align/pdk/finfet/transistor_array.py

```diff
@@ -17,17 +18,22 @@ def __init__(self, *args, **kwargs):
 
         super().__init__()
 
-        self.style = None
+        self.patterns = None
         self.PARTIAL_ROUTING = False
+        self.single_device_connect_m1 = True
         for const in self.primitive_constraints:
             if const.constraint == 'generator':
                 if const.parameters is not None:
-                    self.style = const.parameters.get('style')
+                    self.patterns = const.parameters.get('patterns')
                     self.PARTIAL_ROUTING = const.parameters.get('PARTIAL_ROUTING', False)
+                    self.single_device_connect_m1 = const.parameters.get('single_device_connect_m1', True)
```

Comment:
`join_m1` is shorter than `single_device_connect_m1` and can be used irrespective of number of devices.